onc-healthit/inferno-program
# Upgrade Ruby to 2.7.2, and thin to 1.8.0

# Summary
Ruby 2.5.6 has reached end-of-support, so it's time to upgrade to a newer version. 2.7.x is our choice, because it is compatible with Inferno with very minor modifications (in the JSON gem), and is a stepping-stone to Ruby 3.

## New behavior
* Update .ruby-version, .tool-versions, and Dockerfile to specify Ruby 2.7.2, the latest patch version on the 2.7 branch.
* Add a patch for a deprecation warning in the `json_pure` gem. We can't upgrade `json_pure`, because the old version is locked in `DataMapper`, so this is a workaround.
* Update `thin` to `1.8.0` for Big Sur compatibility

## Code changes
See above

## Testing guidance
* `bundle exec rake test`
* Run docker production and dev environments using the relevant commands
* Ensure `warning: Using the last argument as keyword parameters is deprecated` does not show up in STDOUT after `bundle exec rackup` locally. See See https://github.com/flori/json/issues/399 for more info on that deprecation warning.
